homeassistant/components/plex/config_flow.py

```diff
@@ -0,0 +1,158 @@
+"""Config flow for Plex."""
+import logging
+
+import plexapi.exceptions
+import requests.exceptions
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_URL, CONF_TOKEN, CONF_SSL, CONF_VERIFY_SSL
+from homeassistant.util.json import load_json
+
+from .const import (  # pylint: disable=unused-import
+    CONF_SERVER,
+    CONF_SERVER_IDENTIFIER,
+    DEFAULT_VERIFY_SSL,
+    DOMAIN,
+    PLEX_CONFIG_FILE,
+    PLEX_SERVER_CONFIG,
+)
+from .errors import NoServersFound, ServerNotSpecified
+from .server import PlexServer
+
+USER_SCHEMA = vol.Schema({vol.Required(CONF_TOKEN): str})
+
+_LOGGER = logging.getLogger(__package__)
+
+
+class PlexFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Plex config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize the Plex flow."""
+        self.current_login = {}
+        self.available_servers = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=USER_SCHEMA, errors={}
+            )
+
+        return await self.async_step_server_validate(user_input)
+
+    async def async_step_server_validate(self, server_config):
+        """Validate a provided configuration."""
+        errors = {}
+        self.current_login = server_config
+
+        plex_server = PlexServer(server_config)
+        try:
+            await self.hass.async_add_executor_job(plex_server.connect)
+        except NoServersFound:
```

Comment:
Maybe you can group these exceptions in a better way, those that set errors, those that call a different method, etc...